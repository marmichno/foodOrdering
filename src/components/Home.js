import {Link} from 'react-router-dom';
import {motion} from 'framer-motion';

const containerVariants = {
    hidden: {
        x: 1000
    },
    visible: {
        x: 0,
        transition: {
            type: 'spring',
            stiffness: 120,
            mass: 0.4,
            damping: 8,
            staggerChildren:0.4,
            when:'beforeChildren'
        }
    },
    exit: {
        x:'-100vw',
        transition: {ease: 'easeInOut', duration:0.5}
    }
}

const childVariants = {
    hidden: {
        x: 1000
    },
    visible: {
        x: 0,
        transition: {
            type: 'spring',
            stiffness:60
        }
    }
}

const buttonVariants = {
    hover: {
        scale: 1.3,
        transition: {
            yoyo: Infinity,
            duration: 0.5,
            type: 'tween'
        }
    }
}

export const Home = () => {
    
    return(
        <div className="homeMainContainer">
            {/* <video src="/Videos/video.mp4" muted autoPlay></video>
            <div className="homeMainContainerOverlay"></div> */}
            <motion.div 
            variants={containerVariants}
            initial="hidden"
            animate="visible"
            exit="exit"
            className="homeContentContainer">
                <motion.h1 variants={childVariants}>Freshly made sushi</motion.h1>
                <motion.h1 variants={childVariants}>Delicious ramen</motion.h1>
                <motion.h1 variants={childVariants}>Crispy prawns</motion.h1>
                <motion.h1 variants={childVariants}>Made with passion</motion.h1>
                <motion.hr variants={childVariants}></motion.hr>
                <motion.p variants={childVariants}>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy 
                text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                Type and scrambled it to make a type specimen book.</motion.p>
                <motion.div className="homeButtons" variants={childVariants}>
                    <Link to={{
                        pathname:'/order'
                    }}>
                        <button>Order Online</button>
                    </Link>
                    <Link to={{
                        pathname:'/contact_us'
                    }}>
                        <button>Contact Us</button>
                    </Link>
                </motion.div>
            </motion.div>
         </div>
    )
}