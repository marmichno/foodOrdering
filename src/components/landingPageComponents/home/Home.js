import {Link} from 'react-router-dom';
import {motion} from 'framer-motion';

const containerVariants = {
    hidden: {
        x: 1000
    },
    visible: {
        x: 0,
        transition: {
            type: 'spring',
            stiffness: 120,
            mass: 0.4,
            damping: 8,
            staggerChildren:0.4,
            when:'beforeChildren'
        }
    },
    exit: {
        x:'-100vw',
        transition: {ease: 'easeInOut', duration:0.5}
    }
}

const childVariants = {
    hidden: {
        x: 1000
    },
    visible: {
        x: 0,
        transition: {
            type: 'spring',
            stiffness:60
        }
    }
}

const buttonVariants = {
    initial:{
        scale:1
    },
    hover: {
        scale: 0.9,
        transition: {
            yoyo:Infinity,
            duration:0.5
        }
    }
}

export const Home = () => {
    
    return(
        <div className="homeMainContainer">
            {/* <video src="/Videos/video.mp4" muted autoPlay></video>
            <div className="homeMainContainerOverlay"></div> */}
            <p>swag</p>
            <motion.div 
            variants={containerVariants}
            initial="hidden"
            animate="visible"
            exit="exit"
            className="homeContentContainer">
                <motion.h1 variants={childVariants}>Freshly made sushi</motion.h1>
                <motion.h1 variants={childVariants}>Delicious ramen</motion.h1>
                <motion.h1 variants={childVariants}>Crispy prawns</motion.h1>
                <motion.h1 variants={childVariants}>Made with passion</motion.h1>
                <motion.hr variants={childVariants}></motion.hr>
                <motion.p variants={childVariants} data-testid="pText">Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy 
                text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                Type and scrambled it to make a type specimen book.</motion.p>
                <motion.div className="homeButtons" variants={childVariants}>
                    <Link to={{
                        pathname:'/order'
                    }}>
                        <button data-testid="btnOrderOnline">Order Online</button>
                    </Link>
                    <Link to={{
                        pathname:'/contact_us'
                    }}>

                        <button data-testid="btnContactUs">Contact Us</button>

                    </Link>
                </motion.div>
            </motion.div>
         </div>
    )
}