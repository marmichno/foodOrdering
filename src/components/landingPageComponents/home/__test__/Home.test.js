import {Home} from '../Home';
import {BrowserRouter} from 'react-router-dom';
import TestRenderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

test("matches snapshot", () => {
    const tree = TestRenderer.create(<BrowserRouter><Home/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});