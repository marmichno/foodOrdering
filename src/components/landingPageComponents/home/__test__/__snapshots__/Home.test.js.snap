// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="homeMainContainer"
>
  <p>
    swag
  </p>
  <div
    className="homeContentContainer"
    style={
      Object {
        "transform": "translateX(1000px) translateZ(0)",
      }
    }
  >
    <h1
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      Freshly made sushi
    </h1>
    <h1
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      Delicious ramen
    </h1>
    <h1
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      Crispy prawns
    </h1>
    <h1
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      Made with passion
    </h1>
    <hr
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    />
    <p
      data-testid="pText"
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. Type and scrambled it to make a type specimen book.
    </p>
    <div
      className="homeButtons"
      style={
        Object {
          "transform": "translateX(1000px) translateZ(0)",
        }
      }
    >
      <a
        href="/order"
        onClick={[Function]}
      >
        <button
          data-testid="btnOrderOnline"
        >
          Order Online
        </button>
      </a>
      <a
        href="/contact_us"
        onClick={[Function]}
      >
        <button
          data-testid="btnContactUs"
        >
          Contact Us
        </button>
      </a>
    </div>
  </div>
</div>
`;
